plugins/module_utils/ec2.py

```diff
@@ -92,6 +101,881 @@
+def delete_route(client, route_table_id: str, **params: Dict[str, str]) -> bool:
+    client.delete_route(RouteTableId=route_table_id, **params)
+    return True
+
+
+@EC2VpcRouteTableRouteErrorHandler.common_error_handler("replace route")
+@AWSRetry.jittered_backoff()
+def replace_route(client, route_table_id: str, **params: Dict[str, str]) -> None:
+    client.replace_route(RouteTableId=route_table_id, **params)
+
+
+@EC2VpcRouteTableRouteErrorHandler.common_error_handler("create route")
+@AWSRetry.jittered_backoff()
+def create_route(client, route_table_id: str, **params: Dict[str, str]) -> None:
+    client.create_route(RouteTableId=route_table_id, **params)
+
+
+# EC2 VPC
+class EC2VpcErrorHandler(AWSErrorHandler):
+    _CUSTOM_EXCEPTION = AnsibleEC2Error
+
+    @classmethod
+    def _is_missing(cls):
+        return is_boto3_error_code("InvalidVpcID.NotFound")
+
+
+@EC2VpcErrorHandler.list_error_handler("describe vpcs", [])
+@AWSRetry.jittered_backoff()
+def describe_vpcs(client, **params: Dict[str, Any]) -> List[Dict[str, Any]]:
+    paginator = client.get_paginator("describe_vpcs")
+    return paginator.paginate(**params).build_full_result()["Vpcs"]
+
+
+@EC2VpcErrorHandler.deletion_error_handler("delete vpc")
+@AWSRetry.jittered_backoff()
+def delete_vpc(client, vpc_id: str) -> bool:
+    client.delete_vpc(VpcId=vpc_id)
+    return True
+
+
+@EC2VpcErrorHandler.common_error_handler("describe vpc attribute")
+@AWSRetry.jittered_backoff()
+def describe_vpc_attribute(client, vpc_id: str, attribute: str) -> Dict[str, Any]:
+    return client.describe_vpc_attribute(VpcId=vpc_id, Attribute=attribute)
+
+
+@EC2VpcErrorHandler.common_error_handler("modify vpc attribute")
+@AWSRetry.jittered_backoff()
+def modify_vpc_attribute(client, vpc_id: str, **params) -> None:
+    client.modify_vpc_attribute(VpcId=vpc_id, **params)
+
+
+@EC2VpcErrorHandler.common_error_handler("create vpc")
+@AWSRetry.jittered_backoff()
+def create_vpc(client, **params) -> Dict[str, Any]:
+    return client.create_vpc(**params)["Vpc"]
+
+
+@EC2VpcErrorHandler.common_error_handler("associate vpc cidr block")
+@AWSRetry.jittered_backoff()
+def associate_vpc_cidr_block(client, vpc_id: str, **params) -> None:
+    client.associate_vpc_cidr_block(VpcId=vpc_id, **params)
+
+
+@EC2VpcErrorHandler.common_error_handler("disassociate vpc cidr block")
+@AWSRetry.jittered_backoff()
+def disassociate_vpc_cidr_block(client, association_id: str) -> None:
+    client.disassociate_vpc_cidr_block(AssociationId=association_id)
+
+
+# EC2 Internet Gateway
+class EC2InternetGatewayErrorHandler(AWSErrorHandler):
+    _CUSTOM_EXCEPTION = AnsibleEC2Error
+
+    @classmethod
+    def _is_missing(cls):
+        return is_boto3_error_code("InvalidInternetGatewayID.NotFound")
+
+
+@AWSRetry.jittered_backoff()
+def describe_internet_gateways(client, **params: Dict[str, Any]) -> List[Dict[str, Any]]:
+    return client.describe_internet_gateways(**params)["InternetGateways"]
```

Comment:
Should this use the paginator? There are a bunch of `describe_*` methods in this module with paginators that aren't used here and I think the boto3 recommendation is to use them wherever they exist. If there's a reason we aren't using the paginator for a given method, I think we should note that in a comment.

Reply:
Updated